### DISCUSSION TITLE
Content to multiple files with key as name and content as content

### DISCUSSION THREAD

**Discussion by dtcopcp** [mikefarah/yq#2072] [General]
This feels like it might be trivial, apologies if it is.

If I can filter down to a map in the form:
```
filename1.json: {"some":"json"}
filename2.json: {"some":"json"}
```

how can I split this into two files: `filename1.json` and `filename2.json` with only the json as content?

Thanks

**Comment by mikefarah** [comment]
Sure, you can do it like this:

```
yq '.[]' -s 'key' data1.yaml
```

Explanation:
- `.[]` is the expression we use to split the file contents into the two map values (e.g. `{"some":"json"}`)
- Then we use the split argument `-s` and pass that the expression `key`, letting it know to use the key of each value as the filename

See https://mikefarah.gitbook.io/yq/usage/split-into-multiple-files#split-single-document-into-files for more
